curp/src/server/raw_curp/mod.rs

```diff
@@ -99,6 +98,19 @@ pub(super) struct RawCurp<C: Command, RC: RoleChange> {
     shutdown_trigger: shutdown::Trigger,
 }
 
+impl<C: Command, RC: RoleChange> Debug for RawCurp<C, RC> {
```

Comment:
The supertrait `Debug` has been removed on RoleChange, Without implementing `Debug` in `RC`, RawCurp will not implement it either. This is not what we expect, so I manually made RawCurp implement `Debug`.

Reply:
But `Debug` has been manually implemented on the `Context`, here we can derive `Debug` directly for `RawCurp`

Reply:
Same error in https://github.com/xline-kv/Xline/pull/585#discussion_r1442480594

```
error[E0277]: `RC` doesn't implement `std::fmt::Debug`
   --> curp/src/server/curp_node.rs:878:32
    |
878 |             .field("raw_curp", &self.curp)
    |                                ^^^^^^^^^^ `RC` cannot be formatted using `{:?}` because it doesn't implement `std::fmt::Debug`
    |
note: required for `server::raw_curp::RawCurp<C, RC>` to implement `std::fmt::Debug`
   --> curp/src/server/raw_curp/mod.rs:86:10
    |
86  | #[derive(Debug)]
    |          ^^^^^ unsatisfied trait bound introduced in this `derive` macro
    = note: 1 redundant requirement hidden
    = note: required for `std::sync::Arc<server::raw_curp::RawCurp<C, RC>>` to implement `std::fmt::Debug`
    = note: required for the cast from `std::sync::Arc<server::raw_curp::RawCurp<C, RC>>` to the object type `dyn std::fmt::Debug`
    = note: this error originates in the derive macro `Debug` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting this bound
    |
875 | impl<C: Command, RC: RoleChange + std::fmt::Debug> Debug for CurpNode<C, RC> {
```